### DISCUSSION TITLE
Add support for compose.override.yml files

### DISCUSSION THREAD

**Discussion by troycarpenter** [louislam/dockge#278] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature

### 🔖 Feature description

Have the editor recognize and support editing of compose.override.yml (and docker-compose.override) files.

### ✔️ Solution

Some of my docker images suggest the use of a compose.override.yml file to override the defaults in compose.yml.  It would be nice to have the GUI recognize the files and allow them to be edited as well. 

### ❓ Alternatives

Currently I have to log into the system console and edit those files via the CLI.

**Comment by Kitof** [comment]
It will be very useful. I can't manage mailcow because of it, and that's the only stack missing.

**Comment by R0GGER** [comment]
+1

**Comment by Wigsinator** [comment]
I'd appreciate this, as I use compose override files for things like metadata or service specific information. I'd really appreciate being able to put the x-dockge-url information in the override file instead of cluttering the main file.

**Comment by zndrr** [comment]
+1 for desire of feature. Dockge is otherwise a great tool.

**Comment by scelfo** [comment]
https://github.com/louislam/dockge/discussions/639 is a generic request to add arbitrary file editing support and could be used to satisfy this feature request.

**Reply by nvlbravo** [reply]
I left a comment in the linked thread but in case anyone else is looking into this issue specifically, here's what I am proposing over there:
> <...>
>
> What about replacing or adding something below the default "ENV File" section with something to allow you to configure add-on files as needed, such as:
> ![image](https://private-user-images.githubusercontent.com/25460349/407417711-9c577019-e7a4-41b6-8882-bf5d60f476db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgxMDc1MjAsIm5iZiI6MTczODEwNzIyMCwicGF0aCI6Ii8yNTQ2MDM0OS80MDc0MTc3MTEtOWM1NzcwMTktZTdhNC00MWI2LTg4ODItYmY1ZDYwZjQ3NmRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI4VDIzMzM0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY0OTY3OTczYjJhODhlZDRmYTkxYWIzMjVlNzEzNWQ3NmQ3ZjQzOWM2MGMwMGI4ZGYwNGE4OTBjYmM3YzE1YTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KsvMuo3fpeXSmt5CASavFIVvCyrZCgjyFsE6_dcVyOQ)
> And out pops a:
> ![image](https://private-user-images.githubusercontent.com/25460349/407418142-f69b9722-601c-430e-9626-aa80b5f61f14.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgxMDc1MjAsIm5iZiI6MTczODEwNzIyMCwicGF0aCI6Ii8yNTQ2MDM0OS80MDc0MTgxNDItZjY5Yjk3MjItNjAxYy00MzBlLTk2MjYtYWE4MGI1ZjYxZjE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI4VDIzMzM0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVkNmU1Yjk2N2ViMzhlYzQ4MTExYzFjZjVhYzI2NzViMTc0NDYzNTczOTQ0OGI0OGYyYjZjMWE2ZjcwMWQ1YWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SyCF4w4wL6SEiMR4cWKAvnXjy6vYSd0PuXUhuQtXvsI)
> 
> <...>

I'm envisioning the extension dropdown could include `.Y(A)ML`, and/or `OVERRIDE.Y(A)ML` files and work similarly to potentially meet this request as described.

**Comment by eliasfloreteng** [comment]
I implemented this in #872 and you can use the image from my fork if you change `louislam/dockge:1` to `ghcr.io/eliasfloreteng/dockge:compose-override-editor` in your docker compose.
